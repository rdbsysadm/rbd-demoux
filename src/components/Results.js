import PropTypes from "prop-types";
import React from "react";
import { MDBRow ,MDBCol} from 'mdbreact';

function Results({ children }) {
  return  <MDBRow>    
              <MDBCol  md="6"  xl="4"  sm="3">  
                {children} 
              </MDBCol>    
              <MDBCol   md="6"  xl="4" sm="3">  
                {children} 
              </MDBCol>   
              <MDBCol  md="0" xl="4" sm="3">  
                {children} 
              </MDBCol>   
          </MDBRow>;
}

Results.propTypes = {
  children: PropTypes.node.isRequired
};

export default Results;
