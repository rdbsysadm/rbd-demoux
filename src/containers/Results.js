import PropTypes from "prop-types";
import React, { Component } from "react";

import { withSearch } from "../search-lib";
import { Result, Results } from "../components";
import * as Config from "../config/config-helper";

function capitalizeFirstLetter(string) {
  return string.charAt(0).toUpperCase() + string.slice(1);
}

function htmlEscape(str) {
  return String(str)
    .replace(/&/g, "&amp;")
    .replace(/"/g, "&quot;")
    .replace(/'/g, "&#39;")
    .replace(/</g, "&lt;")
    .replace(/>/g, "&gt;");
}

/*
  Our `Result` component expects result fields to be formatted in an object
  like:
  {
    field1: "value1",
    field2: "value2"
  }
*/
function formatResultFields(result) {
  return Object.keys(result.data).reduce((acc, n) => {
    // Fallback to raw values here, because non-string fields
    // will not have a snippet fallback. Raw values MUST be html escaped.
    let value = result.getSnippet(n) || htmlEscape(result.getRaw(n));
    value = Array.isArray(value) ? value.join(", ") : value;
    acc[`${capitalizeFirstLetter(n)}`] = value;
    return acc;
  }, {});
}
export class ResultsContainer extends Component {
  static propTypes = {
    results: PropTypes.arrayOf(
      PropTypes.shape({
        data: PropTypes.object.isRequired,
        getRaw: PropTypes.func.isRequired,
        getSnippet: PropTypes.func.isRequired
      })
    ).isRequired,
    trackClickThrough: PropTypes.func.isRequired
  };

  handleClickLink = id => {
    const { trackClickThrough } = this.props;
    trackClickThrough(id);
  };

  render() {
    const { results } = this.props;

    return (
      <Results>
        {results.map(result => (
          <Result
            fields={Config.stripUnnecessaryResultFields(
              formatResultFields(result)
            )}
            key={`result-${result.getRaw("id")}`}
            onClickLink={() => this.handleClickLink(result.getRaw("id"))}
            title={Config.getResultTitle(result)}
            url={Config.getResultUrl(result)}
          />
        ))}
      </Results>
    );
  }
}

export default withSearch(ResultsContainer);
