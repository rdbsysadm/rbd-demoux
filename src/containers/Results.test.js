import React from "react";
import { ResultsContainer } from "./Results";
import { shallow } from "enzyme";

const params = {
  results: [
    {
      data: { foo: "foo" },
      getRaw: val => val,
      getSnippet: val => `<em>${val}</em>`
    },
    {
      data: { bar: "bar" },
      getRaw: val => val,
      getSnippet: val => `<em>${val}</em>`
    }
  ],
  trackClickThrough: jest.fn()
};

beforeEach(() => {
  params.trackClickThrough = jest.fn();
});

it("renders correctly", () => {
  const wrapper = shallow(<ResultsContainer {...params} />);
  expect(wrapper).toMatchSnapshot();
});

it("will call back when a document link is clicked in the view", () => {
  const wrapper = shallow(<ResultsContainer {...params} />);

  wrapper
    .find("Result")
    .at(0)
    .prop("onClickLink")();

  const clickThrough = params.trackClickThrough.mock.calls[0][0];
  expect(clickThrough).toEqual("id");
});
