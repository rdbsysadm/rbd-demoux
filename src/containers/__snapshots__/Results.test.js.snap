// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Results>
  <Result
    fields={
      Object {
        "Foo": "<em>foo</em>",
      }
    }
    key="result-id"
    onClickLink={[Function]}
    title="<em>title</em>"
  />
  <Result
    fields={
      Object {
        "Bar": "<em>bar</em>",
      }
    }
    key="result-id"
    onClickLink={[Function]}
    title="<em>title</em>"
  />
</Results>
`;
